### DISCUSSION TITLE
Face issue after add valueEditor (valueEditor: CustomValueEditor,)

### DISCUSSION THREAD

**Discussion by lytechxdigital** [react-querybuilder/react-querybuilder#754] [Q&A] [answered]
Hello team my code works fine before i added value editor as i added  "customvalueeditor" for get all  word city database from my table ....
but after this i face issue in my other fileds like when i add other filter its not working proper and also i added a video so please look and help me i am new in programing word :) my code is following ..

` const CustomValueEditor = (props) => {
    const { fieldData, value, handleOnChange } = props;
    // console.log(fieldData)
    if (fieldData.datatype === "location") {
      // alert(value)

      const DebounceSelect = ({
        fetchOptions,
        debounceTimeout = 800,
        ...props
      }) => {
        const [fetching, setFetching] = useState(false);
        const [options, setOptions] = useState([]);
        const fetchRef = useRef(0);

        const debounceFetcher = useMemo(() => {
          const loadOptions = (value) => {
            fetchRef.current += 1;
            const fetchId = fetchRef.current;
            setOptions([]);
            setFetching(true);

            fetchOptions(value).then((newOptions) => {
              if (fetchId !== fetchRef.current) {
                return;
              }

              setOptions(newOptions);
              setFetching(false);
            });
          };

          return debounce(loadOptions, debounceTimeout);
        }, [fetchOptions, debounceTimeout]);

        return (
          <Select
            labelInValue
            filterOption={false}
            onSearch={debounceFetcher}
            notFoundContent={fetching ? <Spin size="small" /> : null}
            {...props}
            options={options}
            style={{ width: "100%" }} // Ensure full width
          />
        );
      };
      return (
        <div className="w-auto">
          {/* <Select
              mode="multiple"
              placeholder="Inserted are removed"
              // value={selectedItems}
              onChange={handlelocationvalue}
              style={{ width: '100%' }}
              options={SearchLocation.map((item) => ({
                value: item.permalink,
                label: item.value,
              }))}
              onKeyUp={handleSearchLocation}
            /> */}

          <DebounceSelect
            mode="multiple"
            value={SelectedSearchValue}
            placeholder="Select users"
            // onkeyUp={handleSearchLocation}
            fetchOptions={fetchUserList}
            onChange={(newValue) => {
              setSelectedSearchValue(newValue);
            }}
            style={{ width: "100%" }} // Ensure full width
          />
        </div>
      );
    }

    // Fallback for other datatypes
    return <ValueEditor {...props} />;
  };

  const fields = [
    {
      name: "Headquarters Location",
      label: "Headquarters Location",
      placeholder: "E.g. Japan, Boston, Europe",
      onChange: handleLocationEvent, // Use onChange instead of onkeyup
      inputType: "text",
      operators: [
        { name: "includes", label: "includes any" },
        { name: "not_includes", label: "does not include" },
        { name: "blank", label: "is blank" },
        { name: "includes_all", label: "include all" },
      ],
      values: SelectedSearchpermalink, // Replace with your actual location data
      datatype: "location",
    },
    {
      name: "Industry Groupe",
      label: "Industry Groupe",
      inputType: "text",
      placeholder: "Software, Mobile, Health Care",
      operators: [
        { name: "includes", label: "includes any" },
        { name: "not_includes", label: "does not include" },
        { name: "blank", label: "is blank" },
        { name: "includes_all", label: "include all" },
      ],
      valueEditorType: "multiselect",
      values: industryGroupData,
      datatype: "grouup",
    },
    {
      name: "Industries",
      label: "Industries",
      inputType: "text",
      placeholder: "E.g. SaaS, Android, Cloud Computing, Medical Device",
      operators: [
        { name: "includes", label: "includes any" },
        { name: "not_includes", label: "does not include" },
        { name: "blank", label: "is blank" },
        { name: "includes_all", label: "include all" },
      ],
      valueEditorType: "multiselect",
      values: industryFiltterData,
      datatype: "industry",
    },
    {
      name: "Number Of Funding Rounds",
      label: "Number Of Funding Rounds",
      placeholder: "Enter number (e.g. 3)",
      inputType: "number",
      operators: [
        { name: "gte", label: "greater than or equal to" },
        { name: "between", label: "between" },
        { name: "not_eq", label: "does not equal" },
        { name: "eq", label: "equals" },
        { name: "lte", label: "less than or equal to" },
        { name: "blank", label: "is blank" },
      ],
      datatype: "round",
    },
    {
      name: "Last Funding Amount",
      label: "Last Funding Amount",
      placeholder: "Enter amount (e.g. 1,000,000)",
      operators: [
        { name: "gte", label: "greater than or equal to" },
        { name: "between", label: "between" },
        { name: "not_eq", label: "does not equal" },
        { name: "eq", label: "equals" },
        { name: "lte", label: "less than or equal to" },
        { name: "blank", label: "is blank" },
      ],
      datatype: "amount",
    },
    {
      name: "Last Funding Date",
      label: "Last Funding Date",

      placeholder: "5/15/2023",
      operators: [
        { name: "gte", label: "after" },
        { name: "between", label: "between" },
        { name: "lte", label: "before" },
        { name: "eq", label: "equals" },
        { name: "blank", label: "is blank" },
      ],
      datatype: "date",
    },
  ].map((o) => toFullOption(o));

                <ChakraProvider theme={chakraTheme}>
                  <QueryBuilderAntD>
                    <QueryBuilder
                      fields={fields}
                      query={query}
                      onQueryChange={setQuery}
                      controlClassnames={{
                        queryBuilder: "queryBuilder-branches",
                      }}
                      autoSelectField={false}
                      autoSelectOperator={true}
                      controlElements={{
                        valueEditor: CustomValueEditor,
                        addGroupAction: () => null,
                      }}
                    />
                    {/* <Cascader options={options} onChange={onChange} placeholder="Please select" /> */}
                  </QueryBuilderAntD>
                </ChakraProvider>`

image for refrance : 
![Screenshot at Aug 06 18-02-33](https://github.com/user-attachments/assets/735b17f7-0a1a-4d5c-b8a4-3b4399e7e730)

https://github.com/user-attachments/assets/75b4c815-32d9-4fe4-87d3-9a34fd6fc857

**Comment by jakeboone02** [comment]
Hi, could you recreate the issue with a sandbox or a minimal repo? A basic sandbox using the Chakra package can be cloned from here: https://react-querybuilder.js.org/sandbox?t=chakra.

**Comment by lytechxdigital** [comment]
Hello sir i used antd and i aaded almost code in 

[Click here ](https://codesandbox.io/p/devbox/frosty-moser-qzpdwp?file=%2Fsrc%2FApp.tsx&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clzigrciv0007356if6oojdzo%2522%252C%2522sizes%2522%253A%255B90%252C10%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clzigrciu0002356ir3bkdci0%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clzigrciu0004356i09ncdw2a%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clzigrciu0006356iv2qjg76q%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clzigrciu0002356ir3bkdci0%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clzigrcit0001356i8f5gkwgi%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clzihfin40002356iiiqim0yx%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A22%252C%2522startColumn%2522%253A22%252C%2522endLineNumber%2522%253A22%252C%2522endColumn%2522%253A23%257D%255D%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clziiq2ti0002356od1oij23f%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clzigrciu0002356ir3bkdci0%2522%252C%2522activeTabId%2522%253A%2522clziiq2ti0002356od1oij23f%2522%257D%252C%2522clzigrciu0006356iv2qjg76q%2522%253A%257B%2522id%2522%253A%2522clzigrciu0006356iv2qjg76q%2522%252C%2522activeTabId%2522%253A%2522clziirvbp0079356o2jutqrbx%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clzigrciu0005356i9lzd3ka2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A5174%252C%2522id%2522%253A%2522clziirvbp0079356o2jutqrbx%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clzigrciu0004356i09ncdw2a%2522%253A%257B%2522id%2522%253A%2522clzigrciu0004356i09ncdw2a%2522%252C%2522activeTabId%2522%253A%2522clzihhmvn006s356ifjcfn50a%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clzihgm7x004j356ir4yq0ln0%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522id%2522%253A%2522clzihhmvn006s356ifjcfn50a%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clziirfb0000xdlk0b1x541ly%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Reply by jakeboone02** [reply, marked as answer]
You're declaring a component inside another component. See here: https://react-querybuilder.js.org/docs/tips/common-mistakes#custom-component-as-closure

**Reply by lytechxdigital** [reply]
Thank you so much :) it's appreciated.
